pkg/scheduler/scheduler.go

```diff
@@ -410,10 +414,17 @@ func (e entryOrdering) Less(i, j int) bool {
 	return a.Obj.CreationTimestamp.Before(&b.Obj.CreationTimestamp)
 }
 
-func (s *Scheduler) requeueAndUpdate(log logr.Logger, ctx context.Context, w *workload.Info, message string) {
-	added := s.queues.RequeueWorkload(ctx, w)
-	log.V(2).Info("Workload re-queued", "queuedWorkload", klog.KObj(w.Obj), "queue", klog.KRef(w.Obj.Namespace, w.Obj.Spec.QueueName), "added", added)
-	err := workload.UpdateWorkloadStatus(ctx, s.client, w.Obj, kueue.QueuedWorkloadAdmitted, corev1.ConditionFalse, "Pending", message)
+func (s *Scheduler) requeueAndUpdate(log logr.Logger, ctx context.Context, e entry) {
+	if e.skipped {
+		if !s.queues.AddOrUpdateWorkload(e.Obj) {
```

Comment:
hmm, I think we need to `AddIfNotExist` here, we shouldn't update if there is an existing version because if the workload is already back in the queue, it means we received an update event.

Reply:
Added.